File: src/types/cleanup.ts
export interface MemoryCleanupResult {
  freed_bytes: number
  before_used_bytes: number
  after_used_bytes: number
  total_bytes: number
}

export interface MemoryStatus {
  used_bytes: number
  total_bytes: number
  usage_percent: number
  available_bytes: number
}

export interface ScanCategory {
  id: string
  name: string
  description: string
  size_bytes: number
  file_count: number
  paths: string[]
  safe_to_clean: boolean
}

export interface StorageScanResult {
  categories: ScanCategory[]
  total_size_bytes: number
  scan_duration_ms: number
}

export interface StorageCleanResult {
  freed_bytes: number
  freed_file_count: number
  failed_items: string[]
}

export interface ShaderCacheEntry {
  gpu_vendor: string
  cache_path: string
  size_bytes: number
  description: string
}

export interface ShaderCleanResult {
  freed_bytes: number
  cleaned_entries: string[]
  failed_entries: string[]
}

File: src-tauri/src/models/cleanup.rs
use serde::{Deserialize, Serialize};

/// 内存清理结果
#[derive(Serialize, Deserialize, Clone)]
pub struct MemoryCleanupResult {
    pub freed_bytes: u64,
    pub before_used_bytes: u64,
    pub after_used_bytes: u64,
    pub total_bytes: u64,
}

/// 内存状态（查询时返回）
#[derive(Serialize, Deserialize, Clone)]
pub struct MemoryStatus {
    pub used_bytes: u64,
    pub total_bytes: u64,
    pub usage_percent: f64,
    pub available_bytes: u64,
}

/// 存储扫描类别
#[derive(Serialize, Deserialize, Clone, Debug)]
pub struct ScanCategory {
    pub id: String,
    pub name: String,
    pub description: String,
    pub size_bytes: u64,
    pub file_count: u64,
    pub paths: Vec<String>,
    pub safe_to_clean: bool,
}

/// 存储扫描结果
#[derive(Serialize, Deserialize, Clone)]
pub struct StorageScanResult {
    pub categories: Vec<ScanCategory>,
    pub total_size_bytes: u64,
    pub scan_duration_ms: u64,
}

/// 存储清理结果
#[derive(Serialize, Deserialize, Clone)]
pub struct StorageCleanResult {
    pub freed_bytes: u64,
    pub freed_file_count: u64,
    pub failed_items: Vec<String>,
}

/// 着色器缓存条目
#[derive(Serialize, Deserialize, Clone)]
pub struct ShaderCacheEntry {
    pub gpu_vendor: String,
    pub cache_path: String,
    pub size_bytes: u64,
    pub description: String,
}

/// 着色器清理结果
#[derive(Serialize, Deserialize, Clone)]
pub struct ShaderCleanResult {
    pub freed_bytes: u64,
    pub cleaned_entries: Vec<String>,
    pub failed_entries: Vec<String>,
}
